### PR TITLE
Feat/introduce toml usage

### PR DESCRIPTION
**Pull request by nilsjr**
## Proposed Changes
After checking the build setup I identified some differences in versions of used dependencies.

Example:
```
File, Module, Plugin
build.gradle (maestro-android) -> "com.google.protobuf" version "0.8.18"
build.gradle (maestro-client) -> "com.google.protobuf" version "0.9.1"
```
To fix this issue I want to introduce version catalog (libs.versions.toml) so every dependencies + version is in one place. It is now easier to update & maintain all deps used in this repository.

## Testing
Tested it by running **./gradlew test** , installDist, using cli & maestro studio locally.

## Issues Fixed
see above

### CODE REVIEW

File: maestro-android/build.gradle
Comment (nilsjr) [inline maestro-android/build.gradle:83]: Why was this a requirement?

Comment (dmitry-zaitsev) [reply maestro-android/build.gradle:83]: Compiling Android modules with versions of Java higher than 8 is causing bytecode incompatibility issues on certain devices. We should preserve that behavior

File: .github/workflows/publish-cli.yml
Comment (dmitry-zaitsev) [inline .github/workflows/publish-cli.yml:15]: We would need to stick with Java 8 due to requirements in our Android modules

**Comment by dmitry-zaitsev** [comment]
TOML usage is a neat and welcome change! However, can we stick with Java 8 (or at least address it in a different PR)?

File: maestro-android/build.gradle
Comment (nilsjr) [reply maestro-android/build.gradle]: Which devices? Or do you mean a specific android api level?

File: .github/workflows/publish-cli.yml
Comment (nilsjr) [reply .github/workflows/publish-cli.yml]: Ok I will revert these changes and will do a follow up PR for the java 11 update. It also is required when Android Gradle Plugin is updated to 7.4.0 or higher.
[Android Gradle Plugin](https://developer.android.com/studio/releases/gradle-plugin#compatibility-7-4-0)

**Comment by nilsjr** [comment]
Java 11 branch in fork created. [update-java-11-agp-7.4.1](https://github.com/nilsjr/maestro/tree/feat/update-java-11-agp-7.4.1) depends on toml change so waiting until this is merged.

**Review by dmitry-zaitsev — APPROVED** [review]

**Comment by dmitry-zaitsev** [comment]
Thanks @nilsjr ! Everything works 👍

**Pull request merged**
